tests/shared/PCPIntegrationTestDefs.js

```diff
@@ -0,0 +1,506 @@
+/**
+Shared GPII PCP Integration Test definitions
+
+Copyright 2014, 2017 Raising the Floor - International
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+"use strict";
+
+var fluid = require("infusion"),
+    jqUnit = fluid.registerNamespace("jqUnit"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.tests.pcpIntegration");
+fluid.require("%universal");
+gpii.loadTestingSupport();
+
+gpii.tests.pcpIntegration.baseTestDef = {
+    gradeNames: [
+        "gpii.tests.pcpIntegration.testCaseHolder.common.linux",
+        "gpii.test.common.lifecycleManagerReceiver"
+    ],
+    config: {
+        configName: "gpii.tests.acceptance.linux.builtIn.config",
+        configPath: "%universal/tests/platform/linux/configs"
+    }
+};
+
+fluid.defaults("gpii.tests.pcpIntegration.environmentChangedRequestType", {
+    gradeNames: "kettle.test.request.http",
+    path: "/environmentChanged",
+    method: "PUT"
+});
+
+fluid.defaults("gpii.tests.pcpIntegration.client", {
+    gradeNames: "kettle.test.request.ws",
+    path: "/pcpChannel",
+    port: "{configuration}.options.mainServerPort",
+    events: {
+        connectionSucceeded: null
+    },
+    listeners: {
+        connectionSucceeded: {
+            funcName: "gpii.tests.pcpIntegration.connectionSucceeded",
+            args: ["{arguments}.0"]
+        }
+    }
+});
+
+gpii.tests.pcpIntegration.sendMsg = function (evt, client, path, value) {
+    client.send({
+        path: path,
+        value: value,
+        type: "ADD"
+    });
+    setTimeout(function () { // we need to wait for the changes to actually be applied to the system
+        evt.fire("");
+    }, 500);
```

Comment:
It's the only one that I've really managed to come up with .. perhaps @amb26 has a better suggestion. But since these are all integration tests (i.e. with mocked endpoints), there shouldn't really be any delay in applying the settings so I think 500 ms should be more than enough

Reply:
Well - we absolutely need to fix the architecture such that there is an acknowledgement message once settings have been applied. Right now on line 114 of your PCPChannel.js we invoke an update method which returns a promise, which we then throw the results away.
@jobara is working on a more sound scheme for this but we should at the least return some kind of message via the channel, of some new type representing "preferencesApplied". The current client in the PSP will need to be adjusted slightly to ignore this message. It would help if we reframed all our existing messages to be messages of type "modelChanged" so that this new message type is easier to identify.